src/pose_calculations.cpp

```diff
@@ -102,42 +104,45 @@ pose_t transform_pose(const CoordinateSystem co,
         qy_copy = result.qy;
         qz_copy = result.qz;
 
-        float nose_rot_qw;
-        float nose_rot_qx = 0.;
-        float nose_rot_qy;
-        float nose_rot_qz = 0.;
+        double nose_rot_angle;
+        switch(craft_nose)
+        {
+            // left hand pi/2 rotation around Y axis (up)
+            case ArenaDirection::RIGHT: { nose_rot_angle = -M_PI/2.0; break; }
+            // no change, because this is what we have
+            case ArenaDirection::FAR_SIDE: { nose_rot_angle = 0.0; break; }
+            // right hand pi/2 rotation around Y axis (up)
+            case ArenaDirection::LEFT: { nose_rot_angle = M_PI/2.0; break; }
+            // pi rotation around Y
+            case ArenaDirection::NEAR_SIDE: { nose_rot_angle = M_PI; break; }
+        }
 
-        switch(craft_nose) {
-            case ArenaDirection::RIGHT:
-                // left hand pi/2 rotation around Y axis (up)
-                nose_rot_qw = -sqrt(2.0)/2.0;
-                nose_rot_qy =  sqrt(2.0)/2.0;
-                break;
-            case ArenaDirection::FAR_SIDE:
-                // no change, because this is what we have
-                nose_rot_qw = 1.0;
-                nose_rot_qy = 0.0;
-                break;
-            case ArenaDirection::LEFT:
-                // right hand pi/2 rotation around Y axis (up)
-                nose_rot_qw = sqrt(2.0)/2.0;
-                nose_rot_qy = sqrt(2.0)/2.0;
-                break;
-            case ArenaDirection::NEAR_SIDE:
-                // pi rotation around Y
-                nose_rot_qw =  0.0;
-                nose_rot_qy = -1.0;
+        double co_north_angle;
+        switch(co_north)
+        {
+            case ArenaDirection::RIGHT: { co_north_angle = -M_PI/2.0; break; }
+            case ArenaDirection::FAR_SIDE: { co_north_angle = 0.0; break; }
+            case ArenaDirection::LEFT: { co_north_angle = M_PI/2.0; break; }
+            case ArenaDirection::NEAR_SIDE: { co_north_angle = M_PI; break; }
+            case ArenaDirection::TRUE_NORTH:
+                co_north_angle = true_north_deg * M_PI / 180.0;
+                if (co == CoordinateSystem::NED)
+                    co_north_angle *= -1.0;
                 break;
         }
 
+        float nose_rot_qw = cos((nose_rot_angle - co_north_angle)/2);
+        float nose_rot_qx = 0.;
+        float nose_rot_qy = sin((nose_rot_angle - co_north_angle)/2);
+        float nose_rot_qz = 0.;
+
         // perform nose rotation as quaternion rotation https://gegcalculators.com/quaternion-multiplication-calculator-online/
         // result = q_copy * nose_rot  --> use some library here? does boost have quats?
```

Comment:
I actually like the plain c++ here. Since we're not doing anything too fancy with quaternions it makes sense to keep the dependencies minimal. Otherwise, Eigen::Quaterniond would be an option...